velox/exec/TableScan.cpp

```diff
@@ -72,9 +93,7 @@ RowVectorPtr TableScan::getOutput() {
       // w/o producing a result. In this case we return with the Yield blocking
       // reason and an already fulfilled future.
       curStatus_ = "getOutput: task->shouldStop";
-      if ((driverCtx_->task->shouldStop() != StopReason::kNone) ||
-          ((getOutputTimeLimitMs_ != 0) &&
-           (getCurrentTimeMs() - startTimeMs) >= getOutputTimeLimitMs_)) {
+      if (shouldYield(startTimeMs) || shouldStop()) {
```

Comment:
Feels these 2 should be in one method and `Task::shouldStop` called only once.  There is a possibility that `shouldYield` is false in the beginning and `toYield_` increased before `shouldStop` is invoked.  But it is not a severe problem anyway.

Reply:
I see. Changed to cache shouldStop result then.